worlds/yoshisisland/Rules.py

```diff
@@ -0,0 +1,789 @@
+    set_rule(multiworld.get_location("The Deep, Underground Maze: Flowers", player), lambda state: state.has_all({'Chomp Rock', 'Key', 'Large Spring Ball'}, player))
+    set_rule(multiworld.get_location("The Deep, Underground Maze: Stars", player), lambda state: state.has_all({'Chomp Rock', 'Tulip', 'Key'}, player) or (logic.has_midring(state) and state.has_all({'Key', 'Chomp Rock', 'Large Spring Ball'}, player)))
+    set_rule(multiworld.get_location("The Deep, Underground Maze: Level Clear", player), lambda state: state.has_all({'Chomp Rock', 'Key', 'Large Spring Ball', 'Dashed Platform'}, player))
+
+    set_rule(multiworld.get_location("KEEP MOVING!!!!: Red Coins", player), lambda state: (state.has('Egg Capacity Upgrade', player, 1) or logic.combat_item(state)))
+    set_rule(multiworld.get_location("KEEP MOVING!!!!: Flowers", player), lambda state: state.has_all({'Egg Plant'}, player))
+    set_rule(multiworld.get_location("KEEP MOVING!!!!: Stars", player), lambda state: logic.has_midring(state))
+    set_rule(multiworld.get_location("KEEP MOVING!!!!: Level Clear", player), lambda state: True)
+
+    set_rule(multiworld.get_location("King Bowser's Castle: Red Coins", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68CollectibleRoute(state))
+    set_rule(multiworld.get_location("King Bowser's Castle: Flowers", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68CollectibleRoute(state))
+    set_rule(multiworld.get_location("King Bowser's Castle: Stars", player), lambda state: state.has_all({'Helicopter Morph', 'Egg Plant'}, player) and logic._68Route(state))
+
+    if world.options.extras_enabled == 1:
+        set_rule(multiworld.get_location("Poochy Ain't Stupid: Red Coins", player), lambda state: state.has_all({'Poochy'}, player))
+        set_rule(multiworld.get_location("Poochy Ain't Stupid: Flowers", player), lambda state: state.has_all({'Poochy'}, player))
+        set_rule(multiworld.get_location("Poochy Ain't Stupid: Stars", player), lambda state: state.has('Poochy', player))
+        set_rule(multiworld.get_location("Poochy Ain't Stupid: Level Clear", player), lambda state: state.has('Poochy', player))
+
+        set_rule(multiworld.get_location("Hit That Switch!!: Red Coins", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+        set_rule(multiworld.get_location("Hit That Switch!!: Flowers", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+        set_rule(multiworld.get_location("Hit That Switch!!: Stars", player), lambda state: True)
+        set_rule(multiworld.get_location("Hit That Switch!!: Level Clear", player), lambda state: state.has_all({'Large Spring Ball', '! Switch'}, player))
+
+        set_rule(multiworld.get_location("More Monkey Madness: Red Coins", player), lambda state: True)

```

Comment:
afaik, you don't need to manually create lambdas for access rules that are simply "true"

Reply:
Looked into this, apparently the alternative is deleting the entire line since there has to be a rule defined.

Reply:
Yes, that is what I'm proposing. No need for these lines that do nothing